sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/PrivateService.java

```diff
@@ -137,7 +93,36 @@ public static <T> Optional<T> extractCompasPrivate(TBaseElement baseElement, Cla
      * @throws ScdException throws when inconsistency between types
      */
     public static Optional<TCompasICDHeader> extractCompasICDHeader(TPrivate tPrivate) throws ScdException {
-        return extractCompasPrivate(tPrivate, TCompasICDHeader.class);
+        return Optional.of(tPrivate)
+                .filter(aPrivate -> COMPAS_ICDHEADER.getPrivateType().equals(aPrivate.getType()))
+                .map(TAnyContentFromOtherNamespace::getContent)
+                .map(Collection::stream)
+                .flatMap(contentElementsStream -> contentElementsStream
+                        .filter(JAXBElement.class::isInstance)
+                        .map(JAXBElement.class::cast)
+                        .filter(jaxbElement -> !jaxbElement.isNil())
+                        .map(JAXBElement::getValue)
+                        .map(TCompasICDHeader.class::cast)
+                        .reduce((t, t2) -> {
+                            //Vérification du nombre d'élements de même type dans le contenu du private => est-ce vraiment necessaire ??
+                            if (t.getClass() == t2.getClass()) {
+                                throw new ScdException("Expecting maximum 1 element of type "+COMPAS_ICDHEADER+" in private "+tPrivate.getType()+", but got more");
+                            }
+                            return t;
+                        })
+                );
+    }
+
+    public static Stream<TCompasICDHeader> gettCompasICDHeaders(TIED tied) {
+        return tied.getPrivate()
+                .stream()
+                .filter(tPrivate -> COMPAS_ICDHEADER.getPrivateType().equals(tPrivate.getType()))
+                .map(TAnyContentFromOtherNamespace::getContent)
+                .flatMap(Collection::stream)
+                .filter(JAXBElement.class::isInstance)
+                .map(JAXBElement.class::cast)
+                .map(JAXBElement::getValue)
+                .map(TCompasICDHeader.class::cast);
```

Comment:
TIED inherits from TBaseElement.
```suggestion
        return extractCompasPrivates(tied, TCompasICDHeader.class);
```

If you excpect a single value, you can use  `extractCompasPrivate(tied, TCompasICDHeader.class)` which return an Optional.

Reply:
Thanks, 
I will do that